src/main/java/org/killbill/billing/plugin/vertex/VertexTaxCalculator.java

```diff
@@ -298,7 +300,38 @@ private Collection<InvoiceItem> toInvoiceItems(final UUID invoiceId,
+                                                                                       .withAccountId(taxItem.getAccountId())
+                                                                                       .withChildAccountId(taxItem.getChildAccountId())
+                                                                                       .withStartDate(taxItem.getStartDate())
+                                                                                       .withEndDate(taxItem.getEndDate())
+                                                                                       .withAmount(taxItem.getAmount())
+                                                                                       .withCurrency(taxItem.getCurrency())
+                                                                                       .withDescription(taxItem.getDescription())
+                                                                                       .withSubscriptionId(taxItem.getSubscriptionId())
+                                                                                       .withBundleId(taxItem.getBundleId())
+                                                                                       .withCatalogEffectiveDate(taxItem.getCatalogEffectiveDate())
+                                                                                       .withProductName(taxItem.getProductName())
+                                                                                       .withPrettyProductName(taxItem.getPrettyProductName())
+                                                                                       .withPlanName(taxItem.getPlanName())
+                                                                                       .withPrettyPlanName(taxItem.getPrettyPlanName())
+                                                                                       .withPhaseName(taxItem.getPhaseName())
+                                                                                       .withPrettyPhaseName(taxItem.getPrettyPhaseName())
+                                                                                       .withRate(taxItem.getRate())
+                                                                                       .withLinkedItemId(taxItem.getLinkedItemId())
+                                                                                       .withUsageName(taxItem.getUsageName())
+                                                                                       .withPrettyUsageName(taxItem.getPrettyUsageName())
+                                                                                       .withQuantity(taxItem.getQuantity())
+                                                                                       .withItemDetails(String.valueOf(transactionLineDetailModel.getEffectiveRate()))
```

Comment:
Also, we discussed to save it as json considering we may want to add additional details in future.
Current format won't support that.

Reply:
@vlaskhilkevich This PR was experimental. Feel free to create a separate PR with your changes.